### DISCUSSION TITLE
What is the output unit of the JSON exporter

### DISCUSSION THREAD

**Discussion by joritprins** [hubblo-org/scaphandre#251] [General]
I am using Scaphandre to measure the energy/power consumption of a program. I have made a script that saves the process id and runs scaphandre at the same time as the program. I use the JSON exporter and match the PID with the PID i saved. One of the measurements i got is:
`{
   "exe":"program",
   "pid":3819,
  "consumption":5897885.0,
   "timestamp":1673014148.9140978
}`
But what is the unit of the consumption? In the explenation in the [documentation](https://hubblo-org.github.io/scaphandre-documentation/explanations/how-scaph-computes-per-process-power-consumption.html) all the y-axis in the graphs are labeled as "power consumption", so that would mean the unit is Watts. In the [JSON exporter documentation](https://hubblo-org.github.io/scaphandre-documentation/references/exporter-json.html) is the following line `JSON exporter allows you to output the power consumption data in a json file` so this would also be Watts right?

Thanks for reading and helping :)

**Comment by demeringo** [comment]
Hi @joritprins and thanks for asking !

I am pretty sure that the JSON exporter returns the consumption **in microwatts** (i.e. in the same unit as the consumption metric in the prometheus exporter).
https://hubblo-org.github.io/scaphandre-documentation/references/exporter-prometheus.html#scaph_process_power_consumption_microwatts

The mapping from Scaphandre internal value (explicitly in microwatts) to JSON seems to be done here:
- https://github.com/hubblo-org/scaphandre/blob/50fea42ebd1f8bda9d98e8d27b348c626ed3055d/src/exporters/json.rs#L208 
and
- https://github.com/hubblo-org/scaphandre/blob/50fea42ebd1f8bda9d98e8d27b348c626ed3055d/src/exporters/json.rs#L215

I must agree that the doc is not very clear on this point. Furthermore, the name of the field in JSON is less explicit than the name of metric. We should see if we can improve the doc.

**Comment by joritprins** [comment]
Hi @demeringo , thanks for your quick and clear answer!
